### PR TITLE
test: add tests for date conditions

### PR DESCRIPTION
**Pull request by martinkrulltott**
- New test suite for numeric types
- Extends the testing of booleans

### CODE REVIEW

File: cypress/integration/conditions/boolean_conditions.cy.js [hunk 1]
```diff
@@ -34,58 +34,108 @@ const setUpTable = () => {
...
+        assertChipContainsText(`${dimensionName}: 1 condition`)
+
+        assertTooltipContainsEntries([stageName, /\bYes\b/])
+    })
+
+    it('Not answered selected', () => {
+        addConditions(['Not answered'], dimensionName)
 
         expectTableToMatchRows([
             `${getCurrentYearStr()}-01-01`,
```
Comment (jenniferarnesen) [inline cypress/integration/conditions/boolean_conditions.cy.js:76]: Very minor: what about grabbing the year once, then reusing it throughout the tests? `const currentYear = getCurrentYearStr()`

File: cypress/integration/conditions/date_conditions.cy.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+    clickMenubarUpdateButton()
+
+    expectTableToBeVisible()
+}
+
+const addConditions = (conditions) => {
+    cy.getBySelLike('layout-chip').contains(dimensionName).click()
+    conditions.forEach(({ conditionName, value }) => {
+        cy.getWithDataTest('{button-add-condition}').click()
```
Comment (jenniferarnesen) [inline cypress/integration/conditions/date_conditions.cy.js:47]: getBySel

File: cypress/integration/conditions/date_conditions.cy.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        cy.contains('Choose a condition').click()
+        cy.contains(conditionName).click()
+        if (value) {
+            cy.getWithDataTest('{conditions-modal-content}')
```
Comment (jenniferarnesen) [inline cypress/integration/conditions/date_conditions.cy.js:51]: getBySel

File: cypress/integration/conditions/date_conditions.cy.js [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+            selectEventProgram(ANALYTICS_PROGRAM)
+            openDimension(type)
+
+            cy.getWithDataTest('{button-add-condition}').click()
+            cy.contains('Choose a condition').click()
+
+            TEST_OPERATORS.forEach((operator) => {
+                cy.getBySel(
+                    'dhis2-uicore-select-menu-menuwrapper'
```
Comment (jenniferarnesen) [inline cypress/integration/conditions/date_conditions.cy.js:300]: not a big deal at all, but to increase test readability, would it be possible to set the dataTest value on this component? e.g. `conditions-menu`

File: cypress/integration/conditions/date_conditions.cy.js (hunk 2 above)
Comment (martinkrulltott) [reply cypress/integration/conditions/date_conditions.cy.js]: Done, I replaced all of these for the whole Cypress folder.

File: cypress/integration/period.cy.js [hunk 5]
```diff
@@ -6,7 +6,9 @@ describe('period dimension', () => {
     const TEST_DIM_ID = 'eventDate'
     const TEST_DIM_NAME = 'Event date'
     const TEST_RELATIVE_PERIOD_NAME = 'Last 3 months'
-    const TEST_FIXED_PERIOD_NAME = `January ${getCurrentYearStr()}`
+    const TEST_FIXED_PERIOD_NAME = `January ${currentYear}`
+    const currentYear = getCurrentYearStr()
```
Comment (jenniferarnesen) [inline cypress/integration/period.cy.js:10]: Doesn't this need to be defined before using in line 9?

**Review by jenniferarnesen — APPROVED** [review]

**Pull request merged**